Makefile

```diff
@@ -1,74 +1,130 @@
+# ========================
+# Shared Variables
+# ========================
+BUILD_TYPE ?= debug
+BUILD_DIR := build/$(BUILD_TYPE)
+BUILD_SRC_RUST := build/src/rust_extensions
+CURRENT_BUILD := build/current
+DELTA_DIR := rust_extensions/delta
+DELTA_HEADER := $(DELTA_DIR)/target/cxxbridge/delta/src/lib.rs.h
+DELTA_LIB := $(DELTA_DIR)/target/$(BUILD_TYPE)/libdelta.a
+DUCKDB_DIR := third_party/duckdb
+DUCKDB_LIB := $(DUCKDB_DIR)/build/$(BUILD_TYPE)/src/libduckdb.so
+SRC_DIR := src
+
+# ========================
+# Flags
+# ========================
 export override DEBUG := $(filter debug,$(BUILD_TYPE))
 CARGO_FLAGS := $(if $(DEBUG),,--release)
 MAKEFLAGS := --no-print-directory
 
-.PHONY: .BUILD all release debug clean format install installcheck uninstall \
-        duckdb duckdb-fast clean-duckdb \
-        delta clean-delta format-delta
-
-.BUILD:
+# ========================
+# Phony Targets
+# ========================
+.PHONY: .BUILD all clean clean-all clean-delta clean-duckdb debug delta \
+        duckdb duckdb-fast format format-delta help install installcheck \
+        release uninstall
+
+# ========================
+# Default Target: help
+# ========================
+help:
+	@echo "Usage: make [TARGET]"
+	@echo ""
+	@echo "Default BUILD_TYPE is 'debug'."
+	@echo ""
+	@echo "Available targets:"
+	@echo "  all              Build DuckDB and Delta extension"
+	@echo "  clean            Remove build artifacts"
+	@echo "  clean-all        Remove all build artifacts and clean everything"
+	@echo "  clean-delta      Clean Delta extension artifacts"
+	@echo "  clean-duckdb     Clean DuckDB build artifacts"
+	@echo "  debug            Build in debug mode (BUILD_TYPE=debug)"
+	@echo "  delta            Build Delta extension"
+	@echo "  duckdb           Build DuckDB library"
+	@echo "  duckdb-fast      Install prebuilt DuckDB binary"
+	@echo "  format           Format source files"
+	@echo "  format-delta     Format Delta extension code"
+	@echo "  install          Install built artifacts"
+	@echo "  installcheck     Run regression tests"
+	@echo "  release          Build in release mode (BUILD_TYPE=release)"
+	@echo "  uninstall        Uninstall built artifacts"
+
+# ========================
+# Build Setup
+# ========================
+$(BUILD_DIR):
```

Comment:
I would propose to sort targets within each section in the following order
1. Start with targets that are exposed
2. Followed by internal targets sorted in preorder traversal

For example, here it would be
```
debug
release
all
.BUILD
$(BUILD_DIR)
```
Thoughts?

Reply:
Yes, this is much more ergonomic than a naive alphabetic sort. Thanks!